nf_core/pipeline-template/.github/workflows/ci.yml

```diff
@@ -27,33 +29,59 @@ jobs:
         NXF_VER:
           - "24.04.2"
           - "latest-everything"
+        profile:
+          - "conda"
+          - "docker"
+          - "singularity"
+        test:
+          # TODO nf-core: You can customise CI pipeline run tests as required
+          - "test"
+        isMaster:
+          - {% raw %}${{ github.base_ref == 'master' }}{% endraw %}
+        exclude:
+          - isMaster: false
+            profile: "conda"
+          - isMaster: false
+            profile: "singularity"
     steps:
       - name: Check out pipeline code
         uses: actions/checkout@0ad4b8fadaa221de15dcec353f45205ec38ea70b # v4
 
-      - name: Install Nextflow
+      - name: Set up Nextflow
         uses: nf-core/setup-nextflow@v2
         with:
           version: "{% raw %}${{ matrix.NXF_VER }}{% endraw %}"
 
-      - name: Disk space cleanup
-        uses: jlumbroso/free-disk-space@54081f138730dfa15788a46383842cd2f914a1be # v1.3.1
+      - name: Set up Apptainer
+        if: matrix.profile == 'singularity'
+        uses: eWaterCycle/setup-apptainer@main
 
-      - name: Run pipeline with test data (docker)
-        # TODO nf-core: You can customise CI pipeline run tests as required
-        # For example: adding multiple test runs with different parameters
-        # Remember that you can parallelise this by using strategy.matrix
+      - name: Set up Singularity
+        if: matrix.profile == 'singularity'
         run: |
-          nextflow run ${GITHUB_WORKSPACE} -profile test,docker --outdir ./results
+          mkdir -p $NXF_SINGULARITY_CACHEDIR
+          mkdir -p $NXF_SINGULARITY_LIBRARYDIR
+
+      - name: Set up Miniconda
+        if: matrix.profile == 'conda'
+        uses: conda-incubator/setup-miniconda@a4260408e20b96e80095f42ff7f1a15b27dd94ca # v3
+        with:
+          miniconda-version: "latest"
+          auto-update-conda: true
+          channels: conda-forge,bioconda
 
-      - name: Run pipeline with test data (singularity)
-        # TODO nf-core: You can customise CI pipeline run tests as required
+      - name: Set up Conda
+        if: matrix.profile == 'conda'
         run: |
-          nextflow run ${GITHUB_WORKSPACE} -profile test,singularity --outdir ./results
-        if: "{% raw %}${{ github.base_ref == 'master' }}{% endraw %}"
+          conda clean -a
+          conda install -n base conda-libmamba-solver
+          conda config --set solver libmamba
+          echo $(realpath $CONDA)/condabin >> $GITHUB_PATH
+          echo $(realpath python) >> $GITHUB_PATH
```

Comment:
these shouldn't be needed

Reply:
I just copy pasted from modules

Reply:
so which part should I remove there?

Reply:
https://github.com/nf-core/tools/pull/3214/files#r1796450407